feat: add copy button for install commands

imkenough opened the pull request:
#### Description of Change
I've implemented the copy button feature with the following changes; Added copy buttons next to each installation command in `release.handlebars`, Styled the buttons and command containers in `release.css`, Created a new JavaScript file `copy-command.js` to handle the copy functionality (and added the JavaScript file to the main layout)

## What’s This PR About?
- Added a "Copy" button next to the Electron installation commands.
- This makes it easier for users to copy the command without manually selecting it.

## What’s Changed?
- Updated `release.handlebars` to include a clipboard copy feature.
- Used js to enable one-click copying.
- Added visual feedback when copied.

## Issue Reference
- Fixes #124 

## Screenshots 
before:
![image](https://github.com/user-attachments/assets/dfdd8434-f015-4f6b-8117-88808fe6e5c7)

after:
![image](https://github.com/user-attachments/assets/5d9d7a9a-5dd1-46a0-98d4-1e6b125a22d7)


## How to Test
1. Open the Electron release page.
2. Click the "Copy" button next to an install command.
3. Verify that the command is copied to the clipboard. (Win+V)

## Notes for Reviewers
- Let me know if any improvements are needed!

#### Checklist
<!-- Please confirm the following by changing [ ] to [x]. -->

- [x] This PR was not created with AI. (PRs created mainly with AI will be closed. They waste our team's time. We ban repeat offenders.)

PraveenUppar commented:
Hey, I appreciate your enthusiasm, but I'd like to work on this issue myself since I have reported the issue. I was waiting for the maintainer to assign it to me. Can you please refrain from working on it?

imkenough commented:
> Hey, I appreciate your enthusiasm, but I'd like to work on this issue myself since I have reported the issue. I was waiting for the maintainer to assign it to me. Can you please refrain from working on it?

Understood, I wasn’t aware that reporting an issue automatically reserves it. As far as the concern goes, the solution seems to work, so I’ll be leaving my PR up for the maintainer to review. Thanks for the clarification!

nikwen commented:
> Understood, I wasn’t aware that reporting an issue automatically reserves it.

It does not. Anyone can work on any issue.

The Electron team does not assign issues to non-maintainers.

imkenough commented:
> It does not. Anyone can work on any issue.

Apologies for adding an unnecessary reply to this thread, but I appreciate the clarification. Looking forward to any feedback on my PR. Thanks!

dsanders11 commented on src/static/js/copy-command.js:
This doesn't need to be a separate file, and doesn't need to be added to `main.handlebars` when it's only used in `release.handlebars`. Furthermore this should be simplified to a function set as the `onclick` handler for the buttons, and instead of duplicating the data to copy into `data-command`, it can be read out of the sibling `pre.command` element.

dsanders11 commented on src/views/release.handlebars:
```diff
@@ -1,55 +1,131 @@
+      </div>
+      <div class='command-container'>
+        <pre class='command'>yarn add electron@{{version}}</pre>
+        <button class='copy-button' data-command='yarn add electron@{{version}}'>
```
Add a `title="Copy to Clipboard"` attribute to these buttons so it's clear what the button does.

dsanders11 reviewed (changes requested):
This PR contains a lot of unrelated changes to files in `src/views`, please revert those.

I'm not a fan of the visual effect when clicking the copy button, and overall the CSS changes seem overly complex for what is needed.

imkenough commented:
@dsanders11 I've addressed the requested changes (commit `eb7a3da`). Let me know if anything else needs improvement. Thanks for the review!

changes-
added CSS styles for the command block container with proper positioning, the copy button with hover and copied state, proper spacing and visual feedback
modified the `release.handlebars` template to add a copy button to each command block, include icons for copy and check marks and add a title attribute for accessibility

dsanders11 commented:
@imkenough this still contains unrelated changes in `src/views/release.handlebars`.

imkenough commented:
@dsanders11 I've removed the unrelated changes. Apologies for the oversight. (i've also noticed prettier extension keeps trying to change the formatting; fixed it best to my knowledge) My latest commit [e16a8b4](https://github.com/electron/release-status/pull/128/commits/e16a8b46203e26a3648cf4b8830c96de621f82d5) should have the correct fix.

dsanders11 commented:
@imkenough, I'm still seeing unrelated formatting changes in `src/views/release.handlebars` (the `svg` changes, for example). You'll also need to ensure all commits in this PR are verified (the first is not) as we cannot merge PRs unless they are.

imkenough commented:
@dsanders11 Fixed. As mentioned, Prettier was automatically formatting the code, so I disabled it and manually reapplied the fixes. Apologies for the repeated mistake—I didn’t fully understand the issue at first. I’ve also signed my first commit, so the merging should be possible.

erickzhao commented:
>  As mentioned, Prettier was automatically formatting the code, so I disabled it and manually reapplied the fixes.

We should probably fix this in a separate PR:

* `yarn lint` doesn't apply to `.handlebars` files
* `prettierrc.json` is set to `singleQuote: true`
* Format on Save with the Prettier VSCode extension will still format `.handlebars` files with single quotes, causing the diff.

imkenough commented:
Hey @dsanders11 
I noticed my branch is currently 1 commit behind main. If needed, I’d be happy to rebase or sync with the latest upstream changes- just let me know!
Thanks again for reviewing

erickzhao commented:
@imkenough I think this repo doesn't require all branches to be rebased/up to date, so you're good! 👍

imkenough commented:
Thanks for the clarification @erickzhao

dsanders11 commented on src/views/release.handlebars:
```diff
@@ -61,3 +77,14 @@
 </div>
+<script>
+
+  function copyCommand(button) { const command = button.parentElement.textContent.trim();
+
+  navigator.clipboard.writeText(command).then(() => { const icon = button.querySelector('i');
+
+  icon.className = 'fas fa-check'; button.style.color = '#4CAF50'; setTimeout(() => { icon.className
+
+  = 'fas fa-copy'; button.style.color = ''; }, 1000); }); }
```
This formatting is not readable, can you clean it up?

imkenough replied:
Thanks for pointing that out - I've cleaned up the formatting. Let me know if anything else needs adjusting
